mongo-long-stream-test/src/main/java/com/hazelcast/jet/tests/mongo/stream/MongoLongStreamTest.java

```diff
@@ -0,0 +1,197 @@
+import com.hazelcast.jet.mongodb.MongoSources;
+import com.hazelcast.jet.mongodb.impl.MongoUtilities;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.StreamSource;
+import com.hazelcast.jet.tests.common.AbstractSoakTest;
+import com.hazelcast.logging.ILogger;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import org.bson.Document;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.hazelcast.jet.config.ProcessingGuarantee.EXACTLY_ONCE;
+import static com.hazelcast.jet.core.JobStatus.FAILED;
+import static com.hazelcast.jet.core.JobStatus.RUNNING;
+import static com.hazelcast.jet.tests.common.Util.getJobStatusWithRetry;
+import static com.hazelcast.jet.tests.common.Util.sleepMillis;
+import static com.hazelcast.jet.tests.common.Util.sleepMinutes;
+import static com.hazelcast.jet.tests.mongo.stream.MongoLongStreamTest.MongoClientSupplier.getMongoClient;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class MongoLongStreamTest extends AbstractSoakTest {
+    private static final String MONGO_DATABASE = MongoLongStreamTest.class.getSimpleName();
+    private static final int ASSERTION_RETRY_COUNT = 60;
+    private static final int DEFAULT_SNAPSHOT_INTERVAL = 5000;
+    private static final int ASSERTION_ATTEMPTS = 1200;
+    private static final int ASSERTION_SLEEP_MS = 100;
+    private String mongoConnectionString;
+    private int snapshotIntervalMs;
+
+    public static void main(final String[] args) throws Exception {
+        new MongoLongStreamTest().run(args);
+    }
+
+    private static void assertJobStatusEventually(final Job job) {
+        for (int i = 0; i < ASSERTION_ATTEMPTS; i++) {
+            if (job.getStatus().equals(RUNNING)) {
+                return;
+            } else {
+                sleepMillis(ASSERTION_SLEEP_MS);
+            }
+        }
+        throw new AssertionError("Job " + job.getName() + " does not have expected status: " + RUNNING
+                + ". Job status: " + job.getStatus());
+    }
+
+    private static long getNumberOfCurrentlyProcessedDocs(final HazelcastInstance client, final String clusterName) {
+        final Map<String, Long> latestCounterMap = client.getMap(VerificationProcessor.CONSUMED_DOCS_MAP_NAME);
+        return Optional.ofNullable(latestCounterMap.get(clusterName)).orElse(0L);
+    }
+
+    private static void assertCountEventually(final HazelcastInstance client, final long expectedTotalCount,
+                                              final String clusterName) throws Exception {
+        final Map<String, Long> latestCounterMap = client.getMap(VerificationProcessor.CONSUMED_DOCS_MAP_NAME);
+        for (int i = 0; i < ASSERTION_RETRY_COUNT; i++) {
+            final long actualTotalCount = latestCounterMap.get(clusterName);
+            if (expectedTotalCount == actualTotalCount) {
+                return;
+            }
+            SECONDS.sleep(1);
```

Comment:
I would use `sleepSeconds` to have it unified (as we're using `sleepMillis` in `assertJobStatusEventually`).